core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -104,13 +107,16 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   /**
    * Maps the error value of this effect to the specified constant value.
    */
-  final def asError[E1](e1: => E1): ZIO[R, E1, A] = mapError(new ZIO.ConstFn(() => e1))
+  @silent("never used")
```

Comment:
Can't we turn of warning for unused implicit params with `-Ywarn-unused:params.-implicits`?
Might be a bit nicer than silencing everywhere

Reply:
Yes, I think it would definitely be nice to do something so we don't have to have those annotations everywhere. I think we could either do that or there is an option in the silencer to accept a global regex so we could like always call the parameter `canFail` and silence errors that say that was never used. Personally I think the compiler option is cleaner but I want to make sure other people are good with that because it could potentially be useful in some cases. Though I do see we are already silencing some other warnings about implicit parameters not being used.